javaAdvanced_iteratorsAndComparators/src/exercise/strategypattern/ComparatorByName.java

```diff
@@ -0,0 +1,17 @@
+package exercise.strategypattern;
+
+import java.util.Comparator;
+
+public class ComparatorByName implements Comparator<Person> {
+
+	@Override
+	public int compare(Person first, Person second) {
+		int result = first.getName().length() - second.getName().length();
```

Comment:
Maybe it is good to have checks for null. If first is null or first.getName() is null.

Reply:
fixed